_resources/port1.0/group/cmake-1.1.tcl

```diff
@@ -34,73 +34,253 @@
 # Usage:
 # PortGroup     cmake 1.1
 
-namespace eval cmake {}
+namespace eval cmake {
+    # our directory:
+    variable currentportgroupdir [file dirname [dict get [info frame 0] file]]
+}
 
-options cmake.build_dir cmake.install_prefix cmake.install_rpath
-options cmake.out_of_source cmake.set_osx_architectures
+options                             cmake.build_dir \
+                                    cmake.source_dir \
+                                    cmake.generator \
+                                    cmake.build_type \
+                                    cmake.install_prefix \
+                                    cmake.install_rpath \
+                                    cmake.module_path \
+                                    cmake_share_module_dir \
```

Comment:
Why is this one using an underscore instead of `cmake.share_module_dir`?

Or rather, why turn this into an option instead of keeping this as a simple variable? Will there ever be a need to change this from a Portfile?

Reply:
A number of variables were converted into options variables by explicit request in the original PR, or in the PR that gave rise to the current version of cmake-1.1 . I wasn't overly happy with a number of those changes but had little choice (and standardised on the approach afterwards to prevent further discussion).

This variable is the location where ports install their cmake modules. It's probably not necessary that ports ever change this, it's more intended as a read-only variable. It pre-existed in cmake-1.0.tcl and a few ports use it, hence the spelling with an underscore.

You'll notice there's also `cmake.module_path`, which can be used to specify additional locations to search for cmake modules.